app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/supportingevidence/SupportingEvidenceController.scala

```diff
@@ -93,30 +83,22 @@ class SupportingEvidenceController @Inject() (
     request: RequestWithSessionData[_]
   )(
     f: (
-      UUID,
-      CgtReference,
       SessionData,
       FillingOutReturn,
       SupportingEvidenceAnswers
     ) => Future[Result]
   ): Future[Result] =
     request.sessionData.flatMap(s => s.journeyStatus.map(s -> _)) match {
-      case Some((s, r @ FillingOutReturn(c: SubscribedDetails, _, _, d: DraftReturn))) =>
+      case Some((s, r @ FillingOutReturn(_: SubscribedDetails, _, _, d: DraftReturn))) =>
         d match {
-          case DraftSingleDisposalReturn(i, _, _, _, _, _, _, _, _, maybeSupportingDocumentsAnswers, _, _) =>
-            maybeSupportingDocumentsAnswers.fold[Future[Result]](
-              f(i, c.cgtReference, s, r, IncompleteSupportingEvidenceAnswers.empty)
-            )(f(i, c.cgtReference, s, r, _))
-
-          case DraftMultipleDisposalsReturn(i, _, _, _, _, maybeSupportingDocumentsAnswers, _, _) =>
+          case DraftSingleDisposalReturn(_, _, _, _, _, _, _, _, _, maybeSupportingEvidenceAnswers, _, _) =>
+            maybeSupportingEvidenceAnswers.fold[Future[Result]](
+              f(s, r, IncompleteSupportingEvidenceAnswers.empty)
+            )(f(s, r, _))
+          case DraftMultipleDisposalsReturn(_, _, _, _, _, maybeSupportingDocumentsAnswers, _, _) =>
             maybeSupportingDocumentsAnswers.fold[Future[Result]](
-              f(i, c.cgtReference, s, r, IncompleteSupportingEvidenceAnswers.empty)
-            )(f(i, c.cgtReference, s, r, _))
-
-          case i: DraftSingleIndirectDisposalReturn =>
-            i.supportingEvidenceAnswers.fold[Future[Result]](
-              f(i.id, c.cgtReference, s, r, IncompleteSupportingEvidenceAnswers.empty)
-            )(f(i.id, c.cgtReference, s, r, _))
+              f(s, r, IncompleteSupportingEvidenceAnswers.empty)
+            )(f(s, r, _))
```

Comment:
we need a case for `DraftSingleIndirectDisposalReturn` as well to be able to upload files on a single indirect disposal journey

Reply:
Oh, is that one of the new types you mentioned last week that have been introduced?

Reply:
yeah, to accommodate for the new shape of the return for indirect disposals. There'll probably be another one for mixed use later on

Reply:
fixed